internal/rbd/rbd_util.go

```diff
@@ -532,7 +534,15 @@ func (ri *rbdImage) isInUse() (bool, error) {
 	if err != nil {
 		return false, err
 	}
-	ri.Primary = mirrorInfo.Primary
+
+	if mirrorInfo.State == librbd.MirrorImageEnabled {
+		if !ri.Primary {
+			log.ErrorLog(ctx, "cannot map image %s it is not primary", ri)
+
+			return true, nil
+		}
+		ri.Primary = mirrorInfo.Primary
```

Comment:
Can `isInUse()` be called with `ri.Primary` set to `true`?  Previously, `ri.Primary` was reset unconditionally, with this change it's no longer the case.

Reply:
Removed ri.Primary as it's not used anywhere else except this function.